jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/JaxRsClient.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.jaxrs;
+
+import com.google.common.base.Optional;
+import com.palantir.remoting.ssl.SslConfiguration;
+import java.util.Arrays;
+import java.util.List;
+import javax.net.ssl.SSLSocketFactory;
+
+/**
+ * Static factory methods for producing creating HTTP proxies.
+ * TODO(rfink) Better docs.
+ */
+public final class JaxRsClient {
```

Comment:
Yep, will start with no interceptors, and then can add the minimal amount
of adapter API when required.

On Thu, Jul 14, 2016 at 1:58 PM Joe Ellis <notifications@github.com> wrote:

> In
> jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/JaxRsClient.java
> <https://github.com/palantir/http-remoting/pull/160#discussion_r70852567>:
>
> > + * limitations under the License.
> > + */
> > +
> > +package com.palantir.remoting.jaxrs;
> > +
> > +import com.google.common.base.Optional;
> > +import com.palantir.remoting.ssl.SslConfiguration;
> > +import java.util.Arrays;
> > +import java.util.List;
> > +import javax.net.ssl.SSLSocketFactory;
> > +
> > +/**
> > + * Static factory methods for producing creating HTTP proxies.
> > + * TODO(rfink) Better docs.
> > + */
> > +public final class JaxRsClient {
>
> You end up at least having to write your own Request object and likely
> `Response' objects. Might be worth not exposing interceptors for now?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/160/files/d3f84e31f62b6d1726938b1ad07c6f36ff5e46a7#r70852567>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGOdwVglaWEFYE4vJJU1485UIXmYKS-Yks5qVnjHgaJpZM4JL46I>
> .
>